init_scripts/init-root.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```

Comment:
Please elaborate what this script is doing and why it is needed

Reply:
Disables built-in repositories which seemed to slow down the "dnf update" procedure, adding a faster localized repository (israeli based).
Then the script actually updates the system, and installs system-level tools which can come in handy for the developer (as this cloud-based image is stripped, even "vim" is not included)_

Reply:
There is a discussion over Slack whether to use a Fedora box or Ubuntu. You might want to follow that.

Reply:
[From the Vagrant docs](https://www.vagrantup.com/docs/boxes#official-boxes), the HashiCorp (makers of vagrant) and Bento boxes are the only two recommended box sets.  

I've set up the new `bento/ubuntu-20.04` vm box and the sync folder works perfectly -  no need to re-`vagrant up` to copy new data from the synced folder, and it's already comes with essential tools such as vim. I think we should use this box (or a different long time support box) to help us simplify our configuration at the moment.

Also, this box is [supported by GitHub Actions](https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners) as discussed in slack.

Reply:
From my testing, it has a few benefits, but also a few drawbacks.
Benefits:
- /vagrant works without any need for additional configuration
- Future proof for use with Github Actions

Drawbacks:
- "Heavy" package relative to the "cloud-base" (5min+- to download compared to 10seconds)
- Still has packages missing ("python3-pip" for instance is missing)

Overall, it seems worth the move for the Github Actions, so i'll push an update that uses this, but changes are gonna be made to all of the files included to support this.

Reply:
@11galpid , @lioramilbaum I think we can resolve this conversation for now no?